clients/fides-js/src/lib/cookie.ts

```diff
@@ -241,6 +244,39 @@ export const buildCookieConsentForExperiences = (
   return cookieConsent;
 };
 
+/**
+ * Updates prefetched experience, based on:
+ * 1) experience: pre-fetched experience-based consent configuration that does not contain user preference.
+ * 2) cookie: cookie containing user preference.
+ *
+ * Returns updated experience with user preferences.
+ */
+export const updateExperienceFromCookieConsent = (
+  experience: PrivacyExperience,
+  cookie: FidesCookie,
+  debug: boolean
+): PrivacyExperience => {
+  if (!experience.privacy_notices) {
+    return experience;
+  }
+  const updatedExperience = experience;
+  updatedExperience?.privacy_notices?.forEach((notice) => {
+    if (Object.hasOwn(cookie.consent, notice.notice_key)) {
+      // eslint-disable-next-line no-param-reassign
+      notice.current_preference = transformConsentToFidesUserPreference(
+        Boolean(cookie.consent[notice.notice_key]),
+        notice.consent_mechanism
+      );
+    }
+  });
+  debugLog(
+    debug,
+    `Returning updated pre-fetched experience with user consent.`,
+    experience
+  );
+  return updatedExperience;
+};
```

Comment:
Both this function and the next one deserve some unit tests. Previously `__tests__/lib/cookie.test.ts` had almost 100% coverage...

Reply:
ticket here- https://github.com/ethyca/fides/issues/4011